src/decay_storage.cc

```diff
@@ -0,0 +1,71 @@
+#include "decay_storage.h"
+
+namespace decaystorage {
+
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+DecayStorage::DecayStorage(cyclus::Context* ctx) : cyclus::Facility(ctx) {
+  fuel_tracker.Init({&input}, 100000000);
+
+  input = cyclus::toolkit::ResBuf<cyclus::Material>(true);
+  tritium_storage = cyclus::toolkit::ResBuf<cyclus::Material>(true);
+  helium_storage = cyclus::toolkit::ResBuf<cyclus::Material>(true);
+}
+
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+std::string DecayStorage::str() {
+  return Facility::str();
+}
+
+void DecayStorage::EnterNotify() {
+  cyclus::Facility::EnterNotify(); // call base function first
+  buy_policy.Init(this, &input, std::string("input"), &fuel_tracker).Set(incommod).Start();
+  sell_policy.Init(this, &tritium_storage, std::string("output")).Set(outcommod).Start();
+}
+
+void DecayStorage::RecordInventories(double tritium, double helium) {
+  context()
+      ->NewDatum("StorageInventories")
+      ->AddVal("AgentId", id())
+      ->AddVal("Time", context()->time())
+      ->AddVal("TritiumStorage", tritium)
+      ->AddVal("HeliumStorage", helium)
+
+      ->Record();
+}
+
+void DecayStorage::ExtractHelium(
+    cyclus::toolkit::ResBuf<cyclus::Material>& inventory) {
+  if (!inventory.empty()) {
+    cyclus::Material::Ptr mat = inventory.Pop();
+    cyclus::toolkit::MatQuery mq(mat);
+    
+    cyclus::Material::Ptr helium = mat->ExtractComp(mq.mass(He3_id), He3_comp);
+
+    helium_storage.Push(helium);
+    inventory.Push(mat);
+  }
+}
+
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+void DecayStorage::Tick() {
+  tritium_storage.Decay(context()->time());
+  ExtractHelium(tritium_storage);
+  LOG(cyclus::LEV_INFO2, "Storage") << "Quantity to be offered: " << sell_policy.Limit() << " kg.";
+}
+
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+void DecayStorage::Tock() {
+  if (!input.empty()){
+    tritium_storage.Push(input.Pop());
+  }
```

Comment:
Maybe I've forgotten, but is there a reason for a separate `input` and `tritium_storage`?  Can't we receive material straight into `tritium_storage`?

I guess I can't remember if this is just a design pattern that we've used before, or if there is a technical reason for it?